# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xf86-video-mach64/z-hotfix-mmio-move32.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- xf86-video-mach64-6.10.0/src/atimach64accel.c.vanilla	2026-02-14 17:37:48.330868333 +0100
+++ xf86-video-mach64-6.10.0/src/atimach64accel.c	2026-02-14 17:46:28.519132923 +0100
@@ -932,22 +932,22 @@
 
             switch (iDWord)
             {
-                case  0:  MMIO_MOVE32(pDst +  0, 0, *(pSrc +  0));
-                case  1:  MMIO_MOVE32(pDst +  1, 0, *(pSrc +  1));
-                case  2:  MMIO_MOVE32(pDst +  2, 0, *(pSrc +  2));
-                case  3:  MMIO_MOVE32(pDst +  3, 0, *(pSrc +  3));
-                case  4:  MMIO_MOVE32(pDst +  4, 0, *(pSrc +  4));
-                case  5:  MMIO_MOVE32(pDst +  5, 0, *(pSrc +  5));
-                case  6:  MMIO_MOVE32(pDst +  6, 0, *(pSrc +  6));
-                case  7:  MMIO_MOVE32(pDst +  7, 0, *(pSrc +  7));
-                case  8:  MMIO_MOVE32(pDst +  8, 0, *(pSrc +  8));
-                case  9:  MMIO_MOVE32(pDst +  9, 0, *(pSrc +  9));
-                case 10:  MMIO_MOVE32(pDst + 10, 0, *(pSrc + 10));
-                case 11:  MMIO_MOVE32(pDst + 11, 0, *(pSrc + 11));
-                case 12:  MMIO_MOVE32(pDst + 12, 0, *(pSrc + 12));
-                case 13:  MMIO_MOVE32(pDst + 13, 0, *(pSrc + 13));
-                case 14:  MMIO_MOVE32(pDst + 14, 0, *(pSrc + 14));
-                case 15:  MMIO_MOVE32(pDst + 15, 0, *(pSrc + 15));
+                case  0:  MMIO_OUT32(pDst +  0, 0, *(pSrc +  0));
+                case  1:  MMIO_OUT32(pDst +  1, 0, *(pSrc +  1));
+                case  2:  MMIO_OUT32(pDst +  2, 0, *(pSrc +  2));
+                case  3:  MMIO_OUT32(pDst +  3, 0, *(pSrc +  3));
+                case  4:  MMIO_OUT32(pDst +  4, 0, *(pSrc +  4));
+                case  5:  MMIO_OUT32(pDst +  5, 0, *(pSrc +  5));
+                case  6:  MMIO_OUT32(pDst +  6, 0, *(pSrc +  6));
+                case  7:  MMIO_OUT32(pDst +  7, 0, *(pSrc +  7));
+                case  8:  MMIO_OUT32(pDst +  8, 0, *(pSrc +  8));
+                case  9:  MMIO_OUT32(pDst +  9, 0, *(pSrc +  9));
+                case 10:  MMIO_OUT32(pDst + 10, 0, *(pSrc + 10));
+                case 11:  MMIO_OUT32(pDst + 11, 0, *(pSrc + 11));
+                case 12:  MMIO_OUT32(pDst + 12, 0, *(pSrc + 12));
+                case 13:  MMIO_OUT32(pDst + 13, 0, *(pSrc + 13));
+                case 14:  MMIO_OUT32(pDst + 14, 0, *(pSrc + 14));
+                case 15:  MMIO_OUT32(pDst + 15, 0, *(pSrc + 15));
 
                 default:    /* Muffle compiler */
                     break;
